components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -598,17 +601,22 @@ void AdsServiceImpl::OnInitialize(
     const int32_t result) {
   if (result != ads::Result::SUCCESS) {
     LOG(ERROR) << "Failed to initialize ads";
-    return;
-  }
-
-  is_initialized_ = true;
+    is_initialized_ = false;
+  } else {
+    is_initialized_ = true;
 
-  MaybeViewAd();
+    // OnBraveAdsServiceReady is implemented for Android for now
+    #if defined(OS_ANDROID)
+      auto* display_service_impl =
+          static_cast<NotificationDisplayServiceImpl*>(display_service_);
+      display_service_impl->OnBraveAdsServiceReady(is_initialized_);
```

Comment:
When Android is built use `brave-core` a patch should be created for the Chromium changes, however we should look for a more suitable solution too

Reply:
Yes, we can take care about it when Android is fully switched to `brave-core`.

Reply:
@SergeyZhukovsky @gdregalo Can a ticket please be raised for this so that the work is not forgotten, thanks